src/commands/contract/call_function/as_transaction/mod.rs

```diff
@@ -1,30 +1,71 @@
 use dialoguer::Input;
 
+#[derive(Debug, Clone, Default)]
+pub struct FunctionCallAction {
+    pub contract_account_id: Option<crate::types::account_id::AccountId>,
+    pub function_name: String,
+    pub function_args: Vec<u8>,
+    pub gas: crate::common::NearGas,
+    pub deposit: crate::common::NearBalance,
+}
+
 #[derive(Debug, Clone, interactive_clap::InteractiveClap)]
 #[interactive_clap(context = crate::GlobalContext)]
-pub struct CallFunctionAction {
+pub struct CallFunctionProperties {
     ///What is the contract account ID?
     contract_account_id: crate::types::account_id::AccountId,
     ///What is the name of the function?
     function_name: String,
-    ///Enter arguments to this function
+    #[interactive_clap(arg_enum)]
+    #[interactive_clap(skip_default_input_arg)]
+    ///How do you want to pass the function call arguments?
+    function_args_type: super::call_function_args_type::FunctionArgsType,
+    ///Enter the arguments to this function or the path to the arguments file
     function_args: String,
-    #[interactive_clap(long = "prepaid-gas")]
+    #[interactive_clap(named_arg)]
+    ///Enter gas for function call
+    prepaid_gas: super::as_transaction::PrepaidGas,
+}
+
+impl CallFunctionProperties {
+    fn input_function_args_type(
+        _context: &crate::GlobalContext,
+    ) -> color_eyre::eyre::Result<super::call_function_args_type::FunctionArgsType> {
+        super::call_function_args_type::input_function_args_type()
+    }
+
+    pub async fn process(&self, config: crate::config::Config) -> crate::CliResult {
+        let function_args: Vec<u8> = match self.function_args_type {
+            super::call_function_args_type::FunctionArgsType::FileArgs => {
+                let data_path = std::path::PathBuf::from(self.function_args.clone());
+                std::fs::read(data_path).map_err(|err| {
+                    color_eyre::Report::msg(format!("Data file access not found! Error: {}", err))
+                })?
+            }
+            _ => self.function_args.clone().into_bytes(),
```

Comment:
* [ ] `JsonArgs` should be ideally validated to be valid JSON input (the code should attempt to covert `serde_json::Value::from_str()` and check the `Result`)
* [ ] `Base64Args` should be decoded (`base64::decode(self.function_args)`)